src/plugins/discover/public/utils/get_valid_filters.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { DataView, DataViewField } from '@kbn/data-views-plugin/common';
+import { Filter } from '@kbn/es-query';
+
+export const getValidFilters = (dataView: DataView, filters: Filter[]): Filter[] => {
+  return filters.map((filter) => {
+    const meta = { ...filter.meta };
+
+    if (filter.query?.script) {
+      const field = dataView.fields.find((f) => f.name === meta.key);
+
+      // We need to disable scripted filters that are invalid for this data view
+      // since we can't guarantee they'll succeed for the current data view and
+      // can lead to runtime errors
+      if (isValidScriptedFilter(filter, field)) {
```

Comment:
I looked into this and I think there's no good way, non-tacky way to detect the scripted field is still valid. I'd suggest to focus on the problem to prevent displaying runtime errors when switching data views. scripted fields are legacy and we favor the runtime field approach. So for this PR I'd suggest to disable scripted filters when switching data views, to keep it simple.